### PR TITLE
Updates cardano-cli to not use PReferenceScript/SReferenceScript ScriptHash field

### PR DESCRIPTION
**Pull request by Swordlash**
# Changelog

```yaml
- description: Updates cardano-cli to not use PReferenceScript/SReferenceScript ScriptHash field
  type:
  - compatible
  - refactoring
```

# Context

See the [upstream PR](https://github.com/IntersectMBO/cardano-api/pull/642).
The `Maybe ScriptHash` field in `P/SReferenceScript` is redundant in `cardano-api`, causing confusion. Nonetheless it's used in `cardano-cli` as a temporary container for user-provided input. This PR refactors it out so the field can be safely removed from upstream.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs [hunk 1]
```diff
@@ -1354,11 +1354,11 @@ createTxMintValue era (val, scriptWitnesses) =
...
  where
   gatherMintingWitnesses
-    :: [ScriptWitness WitCtxMint era]
+    :: [(Maybe PolicyId, ScriptWitness WitCtxMint era)]
     -> [(PolicyId, ScriptWitness WitCtxMint era)]
   gatherMintingWitnesses [] = []
-  gatherMintingWitnesses (sWit : rest) =
-    case scriptWitnessPolicyId sWit of
+  gatherMintingWitnesses ((pid'm, sWit) : rest) =
+    case scriptWitnessPolicyId pid'm sWit of
```
Comment (smelc) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs]:
```suggestion
  gatherMintingWitnesses ((mPid, sWit) : rest) =
    case scriptWitnessPolicyId mPid sWit of
```

Rationale: having a quote in the middle of a variable name is surprising

**Comment by smelc** [comment]
@Swordlash> please format your changes with:

```
fourmolu -i ...modified hs files...
stylish-haskell -i ...modified hs files...
```

(in this order: `fourmolu` first, then `stylish-haskell`)

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs (hunk 1 above)
Comment (Swordlash) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs]: I picked up this style in Standard Chartered (`'m` for `Maybe`, `'e` for `Either`). No issue, I will adhere to suggested style.

Comment (Swordlash) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs]: All updated, squashed and formatted.

**Comment by Jimbo4350** [comment]
@Swordlash thank you for bringing this to our attention. I have been aware of this for some time but haven't been able to dedicate the time to a solution. We're discussing a potential solution that involves modifying the `cardano-api` and avoids introducing `Maybe PolicyId` as it's done here. I've  asked @carbolymer to work on it.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
We are discussing another way to modify `cardano-api` to avoid having to specify `Maybe PolicyId`.

**Comment by Swordlash** [comment]
Thanks @Jimbo4350, lmk if you'd like me to do something, I'm prepared to dedicate some time for it this week (next weeks I'm away).

**Pull request closed without merging**